python/PresentationBuilder/slides/RemarkSlide.py

```diff
@@ -190,6 +189,19 @@ def parse(self, markdown):
       if self.isParamValid(key):
           markdown = key + ':' + self.parameters()[key] + '\n' + markdown
 
+    # Adjust code starting with *
+    # Remark automatically highlights code lines starting with *, this is disabled
+    # by removing the background in the css and adding an extract *, I hope Remark
+    # will get updated so this is not necessary.
+    #
+    # Also, for some reason re.sub doesn't like this re, so I am doing it manually
+    regex = r'```(.*?)```'
+    for m in re.finditer(regex, markdown, re.DOTALL | re.MULTILINE):
+      for line in m.group(1).split('\n'):
+        if line.strip().startswith('*'):
+          print line
```

Comment:
When I did this with sub(), the sub method was never called. You switch to
findall or finditer, with the exact regex and it finds what I expect. I
will push a commit with the sub() stuff.


On Tue, Mar 3, 2015 at 2:04 PM, Cody Permann <notifications@github.com>
wrote:

> In python/PresentationBuilder/slides/RemarkSlide.py
> <https://github.com/idaholab/moose/pull/4756#discussion_r25726321>:
>
> > @@ -190,6 +189,19 @@ def parse(self, markdown):
> >        if self.isParamValid(key):
> >            markdown = key + ':' + self.parameters()[key] + '\n' + markdown
> >
> > +    # Adjust code starting with *
> > +    # Remark automatically highlights code lines starting with *, this is disabled
> > +    # by removing the background in the css and adding an extract *, I hope Remark
> > +    # will get updated so this is not necessary.
> > +    #
> > +    # Also, for some reason re.sub doesn't like this re, so I am doing it manually
> > +    regex = r'```(.*?)```'
> > +    for m in re.finditer(regex, markdown, re.DOTALL | re.MULTILINE):
> > +      for line in m.group(1).split('\n'):
> > +        if line.strip().startswith('*'):
> > +          print line
>
> It is kind of a nasty piece of code. You claim that re.sub doesn't work?
> Did you make sure to capture the return since it doesn't modify the string
> in place? I still fine it odd that it just doesn't work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/4756/files#r25726321>.
>